index.js

```diff
@@ -60,19 +63,55 @@ function create (dir) {
   }
 
   function add (key, cb) {
-    if (typeof key === 'string') key = new Buffer(key, 'hex')
+    key = datKeyAs.buf(key)
     that.emit('add', key)
     keys.put(hypercore.discoveryKey(key).toString('hex'), key, cb)
   }
 
   function remove (key, cb) {
-    if (typeof key === 'string') key = new Buffer(key, 'hex')
+    key = datKeyAs.buf(key)
     that.emit('remove', key)
     keys.del(hypercore.discoveryKey(key).toString('hex'), cb)
   }
 
+  function get (key, cb) {
+    key = datKeyAs.buf(key)
+    var discKey = hypercore.discoveryKey(key).toString('hex')
+    keys.get(discKey, function (err, key) {
+      if (err) return cb(err) // no key found
+
+      key = datKeyAs.str(key)
+      var feed = core.createFeed(key, {
+        storage: storage(path.join(dir, 'data', key.slice(0, 2), key.slice(2) + '.data'))
+      })
+      feed.get(0, function (err, data) {
+        if (err) return cb(err)
+        var content = hyperdriveFeedKey(data)
+        if (content || !feed.blocks) return done(content)
+        feed.get(feed.blocks - 1, function (err, data) {
+          if (err) return cb(err)
+          done(hyperdriveFeedKey(data))
+        })
+      })
+
+      function done (contentKey) {
+        opened[discKey] = (opened[discKey] || 0) + 1
+        if (!contentKey) return cb(null, feed)
```

Comment:
alternatively we could add a setting where you can pass in the metadata, content feed directly in the archive constructor. this might be the best course of action